### ISSUE TITLE
confusion about the final chamfer distance for LDIF result

### ISSUE THREAD

**Issue by Harvey-Mei** [chengzhag/Implicit3DUnderstanding#12]
Hi, chengzhag.
Thanks for you excellent work.
I tried to reproduce the mesh generation part of your work, exactly the LDIF network, and I train this part followed the parameters mentioned in the paper, the best result I got from my reproduction was wicp_avg_chamfer=7.25, which remain a certain gap with the results shown in the paper(6.72). I am not sure if i missed some thing.
Another question. I used the checkpoint file provided in the readme file to get 6.72.  However, the final statistical results of various categories are not quite consistent with those shown in the paper, so I would like to ask if there is any difference between the statistical results in the paper and the statistical methods in the code?  
![微信截图_20210830142401](https://user-images.githubusercontent.com/48432801/131294735-80e780d7-10b0-4151-ab43-73032b7a69e8.png)
![微信截图_20210830142438](https://user-images.githubusercontent.com/48432801/131294746-3a281c89-6b94-428d-93b8-119bfee101bd.png)

**Comment by chengzhag** [comment]
Hi, @Harvey-Mei,

Thanks for your interest in our work!

I think that the training process might introduce some randomness to the results. Maybe try again to see if there is any change to the results? For your reference, here is the [wandb report](https://wandb.ai/pidan1231239/total3dunderstanding/reports/LDIF--Vmlldzo5NzY3MjE?accessToken=a1kefv01hbhhw6fgot8svlgfad68hz6c48ony6pzg08x6lc1tzd2zjzdpdww1q6u) of experiment 20101613380518, including the LR/training/testing curves. The report also includes another experiment 20101510203369 (grey), which is almost identical with 20101613380518 (orange) in terms of settings. However, the results differ a lot since the learning rate in grey drops early than orange.
![image](https://user-images.githubusercontent.com/9084912/131316469-983da3b6-b920-4937-a98f-4be427949613.png)
The report also includes an experiment 20103116530721 which tests the weight trained on watertight meshes in 20101613380518 on non-watertight meshes. It is where we got the results in the paper.

The reason for your second question might be that the ICP algorithm provided by gaps can introduce some randomness in the process of optimizing to align the meshes. For your reference, chamfer_woICP of each category is: 
{'bed': 4.3483425925046015, 'wardrobe': 5.368200843907593, 'bookcase': 4.65738395402351, 'chair': 6.334140533872199, 'sofa': 5.99414974168976, 'table': 15.723936902234081, 'desk': 10.808811716871475, 'misc': 29.429790655298163, 'tool': 4.530272881545347}.

Hope this can help solve your problem.

**Comment by Harvey-Mei** [comment]
Hi, @chengzhag ,
Very grateful for your quick reply!
I'll study the information you gave me. Good luck with your research.

**Comment by chengzhag** [comment]
PS: The watertight process we use introduces thicker object meshes thus can lead to drop on chamfer distance when evaluating on non-watertight meshes.Although this is what we report in the paper, I personally think it would be fair to use our test results on watertight meshes for comparison.
